torch/_dynamo/utils.py

```diff
@@ -94,20 +94,22 @@ def profile_wrapper(*args, **kwargs):
 
 frame_phase_timing = collections.OrderedDict()
 
-curr_frame = 0
-
+TOTAL_FRAME_COUNTER = 0
+torch._logging._internal.register_formatter_prefix_fn(lambda: f"frame {TOTAL_FRAME_COUNTER}: ")
 
 # Note: Called for you by dynamo - you almost never ever want to invoke this yourself.
-def increment_frame():
-    global curr_frame
-    curr_frame = curr_frame + 1
+def increment_frame(frame_state):
+    global TOTAL_FRAME_COUNTER
+    if "_id" not in frame_state:
+        frame_state["_id"] = TOTAL_FRAME_COUNTER
+        TOTAL_FRAME_COUNTER += 1
 
 
 # Note: Called for you by dynamo - you almost never ever want to invoke this yourself.
 def reset_frame_count():
-    global curr_frame
+    global TOTAL_FRAME_COUNTER
     frame_phase_timing.clear()
-    curr_frame = 0
+    TOTAL_FRAME_COUNTER = 0
```

Comment:
I think this is a bad idea. Unless you are also clearing all the caches, you will have `frame_state` floating around in closures which will now overlap with the reset frame counter. Why are we doing `reset_frame_count`? Is it just for stats? I think we should separate stats from debug labeling. There is no good reason to reset debug labeling.

Reply:
I tend to agree, I can delete this.

We also clear frame_state on reset, btw. Let's keep this monotonically increasing then.